src/mars.d

```diff
@@ -1582,6 +1582,26 @@ int main()
     {
         GC.disable();
     }
+    version(D_Coverage)
+    {
+        // for now we need to manually set the source path
+        string dirName(string path, char separator)
+        {
+            for (size_t i = path.length - 1; i > 0; i--)
+            {
+                if (path[i] == separator)
+                    return path[0..i];
+            }
+            return path;
+        }
+        version (Windows)
+            enum sourcePath = dirName(__FILE_FULL_PATH__, `\`);
+        else
+            enum sourcePath = dirName(__FILE_FULL_PATH__, '/');
```

Comment:
> Couldn't you just run the script that parses the coverage results from the src folder so that the relative paths work.

Well the script just runs the normal test_runner without modifications, the only change is that `dmd` has been compiled with `-cov`.
I am not very familiar with the dmd setup, so if you have better ideas, shoot! ;-)